pyspark_ai/pyspark_ai.py

```
@@ -327,14 +333,42 @@ def create_df(
         return self._create_dataframe_with_llm(page_content, desc, columns, cache)
 
     def _get_transform_sql_query_from_agent(
-        self, temp_view_name: str, schema: str, desc: str
+        self, temp_view_name: str, schema: str, sample_rows_str: str, desc: str
     ) -> str:
         llm_result = self._sql_agent.run(
-            view_name=temp_view_name, columns=schema, desc=desc
+            view_name=temp_view_name,
+            columns=schema,
+            sample_rows=sample_rows_str,
+            desc=desc,
         )
         sql_query_from_response = AIUtils.extract_code_blocks(llm_result)[0]
         return sql_query_from_response
 
+    def _convert_row_as_tuple(self, row: Row) -> tuple:
+        return tuple(map(str, row.asDict().values()))
+
+    def _get_dataframe_results(self, df: DataFrame) -> list:
+        return list(map(self._convert_row_as_tuple, df.collect()))
+
+    def _get_sample_spark_rows(self, df: DataFrame, temp_view_name: str) -> str:
+        if self._sample_rows_in_table_info <= 0:
+            return ""
+        columns_str = "\t".join(list(map(lambda f: f.name, df.schema.fields)))
+        try:
+            sample_rows = self._get_dataframe_results(df.limit(3))
+            # save the sample rows in string format
+            sample_rows_str = "\n".join(["\t".join(row) for row in sample_rows])
+        except Exception:
```

Do we want sampling to fail silently?

---

Yes, sample rows are optional